packages/router/src/router.tsx

```diff
@@ -1,34 +1,131 @@
 // The guts of the router implementation.
 
-import React, { ReactChild, ReactElement } from 'react'
-
-import { useAuth as useAuthHook } from '@redwoodjs/auth'
+import React from 'react'
 
 import {
-  Location,
   parseSearch,
   replaceParams,
   matchPath,
-  mapNamedRoutes,
   PageLoader,
   Redirect,
-  LocationContextType,
-  ParamType,
+  useLocation,
+  validatePath,
+  LocationProvider,
 } from './internal'
+import { PrivateContextProvider, usePrivate } from './private-context'
+import {
+  RouterContextProvider,
+  RouterState,
+  useRouterState,
+} from './router-context'
 import { SplashPage } from './splash-page'
+import { flattenAll, isReactElement } from './util'
+
+/**
+ * A more specific interface will be generated by
+ * babel-plugin-redwood-routes-auto-loader when a redwood project is built
+ *
+ * Example:
+ * interface AvailableRoutes {
+ *   home: () => "/"
+ *   test: () => "/test"
+ * }
+ */
+interface AvailableRoutes {
+  [key: string]: (args?: Record<string, string>) => string
+}
+
+const namedRoutes: AvailableRoutes = {}
```

Comment:
```suggestion
// Named routes is populated at run-time by iterating over the `<Route />` components,
// and appending them to this object.
const namedRoutes: AvailableRoutes = {}
```

Reply:
👍 That's a good description, I added it to the code